include/fastrtps/utils/collections/ResourceLimitedContainerConfig.hpp

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file ResourceLimitedContainerConfig.hpp
+ *
+ */
+
+#ifndef FASTRTPS_UTILS_COLLECTIONS_RESOURCELIMITEDCONTAINERCONFIG_HPP_
+#define FASTRTPS_UTILS_COLLECTIONS_RESOURCELIMITEDCONTAINERCONFIG_HPP_
+
+#include <cstddef>
+#include <limits>
+
+namespace eprosima {
+namespace fastrtps {
+
+/**
+ * Specifies the configuration of a resource limited collection.
+ * @ingroup UTILITIES_MODULE
+ */
+struct ResourceLimitedContainerConfig
+{
+
+#if defined(_MSC_VER) && _MSC_VER <= 1900
+
+    // Initializer list construction won't work on MSVS 2015 if you have struct fields initializers
+    ResourceLimitedContainerConfig(
+            size_t ini = 0, 
+            size_t max = std::numeric_limits<size_t>::max(),
+            size_t inc = 1u)
+        : initial(ini)
+        , maximum(max)
+        , increment(inc)
+    {
+    }
+
+    // MSVS 2015 is not able to treat our static inline methods as constexpr
+    #define ___CONSTEXPR___
+#else
+    #define ___CONSTEXPR___ constexpr
```

Comment:
`config.h` already defined `CONSTEXPR`. You can use it if both suppose a duplication.

Reply:
I have removed it completely, as there were problems on [this branch](https://github.com/eProsima/Fast-RTPS/tree/feature/writer-resource-limits) with this.